backend/Types.mo

```diff
@@ -66,7 +76,22 @@ public type ProfileInfoPlus = {
  likedVideos: [VideoInfo];
  hasPic: Bool;
  rewards: Nat;
- abuseFlags: Nat; // abuseFlags counts other users' flags on this profile, for possible blurring.
+ abuseFlagCount: Nat; // abuseFlags counts other users' flags on this profile, for possible blurring.
+ /// abuseFlagFlag is
+ /// ?true if we (the User requesting this profile) has flagged this profile for abuse.
+ /// ?false if not, and
+ /// null if no specific requesting user is defined by context.
+ abuseFlagFlag: ?Bool;
```

Comment:
I find this name a bit confusing with the double Flag. It's an indicator of whether the requesting user has flagged the profile or video, I might call it something like `hasFlagged', but curious to hear your thoughts.

Reply:
I agree that `FlagFlag` is calling out to be renamed.  Thanks for the attention. :)

I think the word "has" in `hasFlagged` is only problematic here (for my reading at least) because the "who" that "has" is a little ambiguous.  The userName in the record is the wrong "who", and that's what the field name might suggest, paired with the (sibling) fieldname `userName`, already there.

So I tried to avoid "has" and got something potentially worse. :)

How about `viewerhasFlagged`?

That's totally clear to me, albeit more verbose.

Reply:
I like `viewerHasFlagged`! Lets do that.